aws-rds-optiongroup/src/main/java/software/amazon/rds/optiongroup/ReadHandler.java

```diff
@@ -0,0 +1,53 @@
+package software.amazon.rds.optiongroup;
+
+import java.util.List;
+
+import software.amazon.awssdk.services.rds.RdsClient;
+import software.amazon.awssdk.services.rds.model.ListTagsForResourceRequest;
+import software.amazon.awssdk.services.rds.model.ListTagsForResourceResponse;
+import software.amazon.awssdk.services.rds.model.OptionGroup;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+public class ReadHandler extends BaseHandlerStd {
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<RdsClient> proxyClient,
+            final Logger logger) {
+
+        return proxy.initiate("rds::read-option-group", proxyClient, request.getDesiredResourceState(), callbackContext)
+                .translateToServiceRequest(Translator::describeOptionGroupsRequest)
+                .backoffDelay(BACKOFF_DELAY)
+                .makeServiceCall((describeOptionGroupsRequest, proxyInvocation) -> {
+                    return proxyInvocation.injectCredentialsAndInvokeV2(describeOptionGroupsRequest, proxyInvocation.client()::describeOptionGroups);
+                })
+                .handleError((describeOptionGroupsRequest, exception, client, resourceModel, ctx) -> {
+                    return handleException(exception);
+                })
+                .done((describeOptionGroupsRequest, describeOptionGroupsResponse, proxyInvocation, model, context) -> {
+                    final OptionGroup optionGroup = describeOptionGroupsResponse.optionGroupsList().stream().findFirst().get();
+                    final List<OptionConfiguration> optionConfigurations = Translator.translateOptionConfigurationsFromSdk(optionGroup.options());
+                    final ListTagsForResourceRequest listTagsForResourceRequest = Translator.listTagsForResourceRequest(optionGroup.optionGroupArn());
+                    final ListTagsForResourceResponse listTagsForResourceResponse = proxyInvocation.injectCredentialsAndInvokeV2(
+                            listTagsForResourceRequest,
+                            proxyInvocation.client()::listTagsForResource
+                    );
```

Comment:
for tagging apis I think we should softfail them. As there are stack level tags that customer might not want to attach to this particular resource. Meaning that role will not have granted permissions to list tags. This is considered a normal customer behavior, so we should use something like this: https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-kms/blob/master/key/src/main/java/software/amazon/kms/key/UpdateHandler.java#L107

Reply:
I hope I have fixed this. Thank you.